src/astarte-charts/react/ConnectedDevices/index.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+   This file is part of Astarte.
+
+   Copyright 2020 Ispirata Srl
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import React, { useMemo } from 'react';
+import ApexChart from 'react-apexcharts';
+import _ from 'lodash';
+import type { AstarteChartProvider } from 'astarte-charts';
+
+import { useChartProviders } from '../hooks';
+
+interface AstarteConnectedDevicesChartProps {
+  providers: AstarteChartProvider[];
```

Comment:
A few ideas about how to improve this kind of scenario:
- introduce checks in each chart to make sure only supported Providers and DataTypes are provided via props
- make Providers return a properly subtyped AstarteChartValue, with more specific properties
- add manual checks in the Chart for this specific case
What do you think?

Reply:
option 1) and 3) while working, make no use of the type system we're using. Since we have typescript by our side, i'd encourage using it, this way other developers will also get notified at compile time, rather than seeing their app crash at runtime

Reply:
On the contrary, using a Typescript only solution will not be of use for non typescript developers.
However I am evaluating better types since I agree with you

Reply:
This is not a valid argument. If they don't use typescript their code will crash at runtime the same way it does right now if you don't use the object that the chart expects